### PR TITLE
Ajay assignment 2

### PR DESCRIPTION
**Pull request by Ajay-Kumar-VS**

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,7 +1,18 @@
+import { SafeAreaView,View,StatusBar } from 'react-native'
```
Comment (pankaj-negi-vs) [inline App.tsx:1]: remove unused imports

File: App.tsx [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
 import { NavigationContainer } from '@react-navigation/native'
 import StackNavigator from './src/navigation/stack-navigator/StackNavigator.tsx'
 
+import Dashboard from './src/screens/dashboard/Dashboard.jsx'
```
Comment (pankaj-negi-vs) [inline App.tsx:5]: the project is in tsx

File: src/assets/images/hero.png
Comment (pankaj-negi-vs) [inline src/assets/images/hero.png:1]: change the name of the file to dashboardBackground

File: src/screens/dashboard/header/Header.tsx
Comment (pankaj-negi-vs) [inline src/screens/dashboard/header/Header.tsx:1]: use prettier

File: src/assets/icons/overthinkerWhite.png
Comment (pankaj-negi-vs) [inline src/assets/icons/overthinkerWhite.png:1]: rename this to overthinkerDashboard

File: src/screens/dashboard/Dashboard.tsx
Comment (pankaj-negi-vs) [inline src/screens/dashboard/Dashboard.tsx:1]: change the folder structure

File: src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
+import { View, Text, StyleSheet, Image, ImageBackground } from 'react-native';
+import React from 'react'
+import { COLORS } from '../../../theme/colors'
+import { Typography } from '../../../theme/typography'
+import { Colors } from 'react-native/Libraries/NewAppScreen'
+import { palette } from '../../../theme/palette'
+
+const completedIcon = require('../../../assets/icons/complete.png')
+const markedFavourite = require('../../../assets/icons/markedFavourite.png')
+const notmMarkedFavourite = require('../../../assets/icons/notMarkedFavourite.png')
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx]: Move these constants in a separate file of dashboard constant

File: src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  [key: string]: string;
+}
+
+const backgroundColorDetails: BackgroundColorDetails = {
+  'D-active': COLORS.secondary[100],
+  'Reminder of the day': COLORS.accent[100],
+  'Reflection of the day': COLORS.primary[100]
+};
+
+interface IChallengeDeatilsCardProps{
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx]: move interface and type in a separate file of types

File: src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  'D-active' = dActive,
+  'Reminder of the day' = reflection,
+  'Reflection of the day' = reminder
+}
+interface BackgroundColorDetails {
+  [key: string]: string;
+}
+
+const backgroundColorDetails: BackgroundColorDetails = {
+  'D-active': COLORS.secondary[100],
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx]: move this to constant

File: src/screens/dashboard/header/Header.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
+
+import { ImageBackground, StatusBar, StyleSheet, Text, View, Image } from 'react-native'
+import { styles } from './header-style'
+import React from 'react'
+import { getTodayDate } from '../../../utils/getTodayDate'
+const headerBackground = require('../../../assets/images/hero.png')
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/header/Header.tsx]: make a separate file in constants to save all the constants there like i have done..... aur dashboard ki constant file should be dashboard-constant.ts fir wha se consts ko import kar

File: src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        endingTime: string,
+        isCompleted: string
+
+}
+const ChallengeDeatilsCard = ({ChallengeDeatils}: IChallengeDeatilsCardProps) => {
+  const {
+    id,
+   title,isFav,startingTime,endingTime,isCompleted
+  } = ChallengeDeatils;
+  console.log(isCompleted===true)
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx]: Why log is not removed

File: src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+  return (
+    <View style={[styles.container,{backgroundColor:backgroundColorDetails[title]}]}>
+      <View style={styles.imageContainer}>
+        <View style={styles.imageBackgroundContainer}>
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx]: why two container?

File: src/screens/dashboard/header/Header.tsx (hunk 6 above)
Comment (pankaj-negi-vs) [inline src/screens/dashboard/header/Header.tsx]: use named exports only

File: src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+          <Image style={styles.overThinkerGrey} source={imageDetails[title]} />
+        </View>
+      </View>
+      <View style={styles.detailsContainer}>
+        <View style={styles.detailsSubContainer}>
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx]: what is this?

File: src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+          <View style={styles.detailsMainContainer}>
+            <View style={styles.details}>
+              <Text style={styles.listenTitle}>Challenge {id}</Text>
+              {/* {console.log(isCompleted)} */}
+              {(isCompleted==='true') ?(<Image style={styles.completedIcon} source={completedIcon} />):(<View/>)}
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx]: isCompleted is enough no need to use ==='true'

File: src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+            </View>
+            <View>
+              {isFav==='true' ? <Image style={styles.starIcon} source={markedFavourite} /> : <Image style={styles.starIcon} source={notmMarkedFavourite} />}
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/ChallengeDeatilsCard.tsx]: same here

File: src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts [hunk 12]
```diff
@@ -0,0 +1,113 @@
+import { StyleSheet } from "react-native";
+import { Typography } from '../../../theme/typography'
+import { COLORS } from '../../../theme/colors'
+import { Spacing } from "../../../theme/spacing";
+import { palette } from "../../../theme/palette";
+
+export const styles = StyleSheet.create({
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts]: where is styles interface

File: src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+    audioIconBackground:{
+        height:36,
+        width:36,
+        borderRadius:65.62,
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts]: Why values are hardcoded

Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts]: never use decimal values

File: src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+        fontSize:12,
+        height:18,
+        letterSpacing:0.02,
+        color:COLORS.neutral[500]
+      },
+      timingDetailsContainer:{
+        justifyContent:"space-between",
+        alignItems:"center",
+        flexDirection:'row',
+        // backgroundColor:"red"
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts]: why comment?

File: src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+        // flex:1,
+        alignItems:"center",
+        justifyContent:'space-between',
+        flexDirection: 'row',
+        // backgroundColor:"yellow"
+    
+      },
+      detailsSubContainer: {
+        // flex:1,
+        // flexDirection:'row',
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts]: empty ??

File: src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+      },
+      imageContainer: {
+          marginVertical:28,
+      },
+      detailsContainer: {
+        flex:1,
+        marginVertical:24,
+        gap:4,
+        // backgroundColor:"blue"
+      },
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/challenge-details-card/challenge-details-card-style.ts]: Order is also wrong if you have 
container 1 -> 2- > 3 then in styles it should also be in this way

File: src/screens/dashboard/dashboard-styles.ts [hunk 17]
```diff
@@ -0,0 +1,20 @@
+import { StyleSheet } from "react-native";
+import { COLORS } from '../../theme/colors';
+
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/dashboard-styles.ts:4]: Why this file is outside of dashboard folder?

File: src/screens/dashboard/header/Header.tsx [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+const calendarImage = require('../../../assets/icons/calendar.png')
+const overthinkerImage = require('../../../assets/icons/overthinkerWhite.png')
+const Header = () => {
+    let date: string = getTodayDate()
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/header/Header.tsx]: make a separate function to calculate the time

File: src/screens/dashboard/Dashboard.tsx [hunk 19]
```diff
@@ -1,14 +1,102 @@
...
+      <MasteryOftheDayCard masteryCardDetails={masteryCardDetails} />
+      </View>
+      <View style={styles.flatListContainer}>
+      <View style={styles.FlatList}>
+      {isLoading ? (
+        <ActivityIndicator />
+      ) :
+      <FlatList 
+          data={ChallengesData}
+          renderItem={({item})=><View style={{justifyContent:'center',gap:16,marginBottom:16,borderRadius:16}}><ChallengeDeatilsCard ChallengeDeatils={item}/></View>}
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/Dashboard.tsx]: inline css?

File: src/screens/dashboard/dashboard-styles.ts [hunk 20]
```diff
@@ -0,0 +1,20 @@
...
+
+export const styles=StyleSheet.create({
+    MasteryOftheDayCard:{
+        backgroundColor:COLORS.primary[50]
+    },
+    FlatList:{
```
Comment (tushar-jha-vs) [inline src/screens/dashboard/dashboard-styles.ts]: change name and name should start with lowercase letter

File: src/screens/dashboard/header/Header.tsx (hunk 6 above)
Comment (tushar-jha-vs) [inline src/screens/dashboard/header/Header.tsx]: const files

File: src/screens/dashboard/header/Header.tsx (hunk 18 above)
Comment (tushar-jha-vs) [inline src/screens/dashboard/header/Header.tsx]: let use nahi hota?

File: src/screens/dashboard/header/header-style.ts [hunk 21]
```diff
@@ -0,0 +1,94 @@
+import { StyleSheet } from "react-native";
+import { Typography } from '../../../theme/typography'
+import { COLORS } from '../../../theme/colors'
+import { Spacing } from "../../../theme/spacing";
+
+export const styles = StyleSheet.create({
+    overThinkerImageContainer:{
+        // marginTop:100,
+        // marginBottom: 24,
+        marginLeft: 224,
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/header/header-style.ts]: ?????

File: src/screens/dashboard/header/header-style.ts [hunk 22]
```diff
@@ -0,0 +1,94 @@
...
+        // backgroundColor:"red"
+
+        // paddingHorizontal:4,
+        // paddingVertical:11,
+      },
+      dayCounterDetails:{
+        color:COLORS.white,
+        fontFamily:Typography.primary.bold
+      },
+      overThinkeText:{
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/header/header-style.ts]: name??

File: src/screens/dashboard/header/header-style.ts [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+        color:COLORS.white,
+    
+        fontFamily:Typography.secondary.bold,
+        fontSize:24,
+        lineHeight:32.4,
+      },
+      container: {
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/header/header-style.ts]: why empty container?

File: src/screens/dashboard/header/header-style.ts [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+          // marginBottom:565
+      },
+      subcontainer: {
+        // height: 60,
+        // marginBottom:565
+        // flex:1,
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/header/header-style.ts]: remove the comments

File: src/screens/dashboard/header/header-style.ts [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+        flexDirection: 'row',
+        // marginLeft: 20,
+        // marginVertical: 19.5,
+        height: 21,
+        gap: Spacing.space_8,
+      },
+      overthinkerDetailsContainer: {
+        flexDirection: 'row',
+        alignContent: 'space-between',
+        marginBottom: 0,
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/header/header-style.ts]: what is the use of this??

File: src/screens/dashboard/masteryOfTheDayCard/MasteryOftheDayCard.tsx [hunk 26]
```diff
@@ -0,0 +1,70 @@
+import { View, Text, StyleSheet, Image } from 'react-native'
+import React from 'react'
+import { COLORS } from '../../../theme/colors'
+import { Typography } from '../../../theme/typography'
+import { Colors } from 'react-native/Libraries/NewAppScreen'
+import { palette } from '../../../theme/palette'
+
+const completedIcon = require('../../../assets/icons/complete.png')
+const markedFavourite = require('../../../assets/icons/markedFavourite.png')
+const notmMarkedFavourite = require('../../../assets/icons/notMarkedFavourite.png')
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/masteryOfTheDayCard/MasteryOftheDayCard.tsx]: correct the name

File: src/screens/dashboard/masteryOfTheDayCard/MasteryOftheDayCard.tsx [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+  return (
+    <View style={styles.container}>
+      <View style={styles.imageContainer}>
+        <View style={styles.imageBackgroundContainer}>
+          <Image style={styles.overThinkerGrey} source={img} />
+        </View>
+      </View>
+      <View style={styles.detailsContainer}>
+        <View style={styles.detailsSubContainer}>
+          <View style={styles.detailsMainContainer}>
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/masteryOfTheDayCard/MasteryOftheDayCard.tsx]: bhai main container subcontainer ke andar kya kar rha hai?

File: src/components/challenge-details-card/ASChallengeDeatilsCard.tsx [hunk 28]
```diff
@@ -0,0 +1,60 @@
+const audio = require('../../assets/icons/audio.png')
```
Comment (pankaj-negi-vs) [inline src/components/challenge-details-card/ASChallengeDeatilsCard.tsx]: put const in a separate file then import them

File: src/components/challenge-details-card/ASChallengeDeatilsCard.tsx [hunk 29]
```diff
@@ -0,0 +1,60 @@
...
+const completedIcon = require('../../assets/icons/complete.png')
+const markedFavourite = require('../../assets/icons/markedFavourite.png')
+const notmMarkedFavourite = require('../../assets/icons/notMarkedFavourite.png')
+import React from 'react'
+import { Image, Text, View } from 'react-native'
+import {
```
Comment (pankaj-negi-vs) [inline src/components/challenge-details-card/ASChallengeDeatilsCard.tsx:5]: import order

File: src/components/challenge-details-card/ASChallengeDeatilsCard.tsx [hunk 30]
```diff
@@ -0,0 +1,60 @@
...
+import { styles } from './asChallenge-details-card-style'
+
+const ChallengeDeatilsCard = ({ ChallengeDeatils }: IChallengeDeatilsCardProps) => {
+  const { id, title, isFav, startingTime, endingTime, isCompleted } = ChallengeDeatils
+
+  return (
+    <View style={[styles.container, { backgroundColor: backgroundColorDetails[title] }]}>
+      <View style={styles.imageContainer}>
+        <View style={styles.imageBackgroundContainer}>
+          <Image style={styles.overThinkerGrey} source={imageDetails[title]} />
```
Comment (pankaj-negi-vs) [inline src/components/challenge-details-card/ASChallengeDeatilsCard.tsx]: change name to backgroundImage

File: src/components/challenge-details-card/ASChallengeDeatilsCard.tsx [hunk 31]
```diff
@@ -0,0 +1,60 @@
...
+        </View>
+      </View>
+      <View style={styles.detailsContainer}>
+        <View style={styles.detailsSubContainer}>
+          <View style={styles.detailsMainContainer}>
```
Comment (pankaj-negi-vs) [inline src/components/challenge-details-card/ASChallengeDeatilsCard.tsx]: change name as main always comes first

File: src/components/challenge-details-card/ASChallengeDeatilsCard.tsx [hunk 32]
```diff
@@ -0,0 +1,60 @@
...
+            <View style={styles.details}>
+              <Text style={styles.listenTitle}>Challenge {id}</Text>
+              {/* {console.log(isCompleted)} */}
```
Comment (pankaj-negi-vs) [inline src/components/challenge-details-card/ASChallengeDeatilsCard.tsx]: don't leave comments

File: src/components/challenge-details-card/asChallenge-details-card-style.ts
Comment (pankaj-negi-vs) [inline src/components/challenge-details-card/asChallenge-details-card-style.ts:1]: change the name of the file to asChallengeDetailsCard-styles.ts

File: src/components/challenge-details-card/asChallenge-details-card-style.ts [hunk 33]
```diff
@@ -0,0 +1,108 @@
...
+    height: Spacing.space_16,
+    width: Spacing.space_16,
+  },
+  completedIcon: {
+    height: Spacing.space_16,
+    width: Spacing.space_16,
+  },
+  listenTitle: {
+    color: COLORS.neutral[500],
+    fontFamily: 'Quicksand-bold',
```
Comment (pankaj-negi-vs) [inline src/components/challenge-details-card/asChallenge-details-card-style.ts]: use typography import for this

File: src/components/challenge-details-card/asChallenge-details-card-style.ts [hunk 34]
```diff
@@ -0,0 +1,108 @@
...
+    fontSize: Spacing.space_12,
+    height: Spacing.space_18,
+    letterSpacing: 0.02,
```
Comment (pankaj-negi-vs) [inline src/components/challenge-details-card/asChallenge-details-card-style.ts]: no need for this

File: src/components/challenge-details-card/asChallenge-details-card-style.ts [hunk 35]
```diff
@@ -0,0 +1,108 @@
...
+  details: {
+    flexDirection: 'row',
+    gap: Spacing.space_8,
+  },
+  detailsMainContainer: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  detailsSubContainer: {},
```
Comment (pankaj-negi-vs) [inline src/components/challenge-details-card/asChallenge-details-card-style.ts]: empty style don't need this

File: src/components/dashboard-header/ASDashboardHeader.tsx [hunk 36]
```diff
@@ -1,18 +1,16 @@
...
-import { styles } from './header-style'
 import React from 'react'
-import { getTodayDate } from '../../../utils/getTodayDate'
-const headerBackground = require('../../../assets/images/hero.png')
-const calendarImage = require('../../../assets/icons/calendar.png')
-const overthinkerImage = require('../../../assets/icons/overthinkerWhite.png')
+import { Image, ImageBackground, Text, View } from 'react-native'
+import { getTodayDate } from '../../utils/common-utils'
+import { styles } from './asDashboard-header-style'
+const headerBackground = require('../../assets/images/dashboardHeaderBackground.png')
```
Comment (pankaj-negi-vs) [inline src/components/dashboard-header/ASDashboardHeader.tsx]: separate file for const

File: src/components/dashboard-header/ASDashboardHeader.tsx [hunk 37]
```diff
@@ -23,7 +21,7 @@ const Header = () => {
             </View>
           </View>
         </View>
-        {/* <View style={styles.overthinkerDetailsContainer}> */}
+
         <View style={styles.overthinkerDetailsSubContainer}>
           <Text style={styles.overThinkeText}>Overthinker</Text>
```
Comment (pankaj-negi-vs) [inline src/components/dashboard-header/ASDashboardHeader.tsx]: spelling mistake

File: src/components/dashboard-header/ASDashboardHeader.tsx [hunk 38]
```diff
@@ -23,7 +21,7 @@ const Header = () => {
             </View>
           </View>
         </View>
-        {/* <View style={styles.overthinkerDetailsContainer}> */}
+
         <View style={styles.overthinkerDetailsSubContainer}>
```
Comment (pankaj-negi-vs) [inline src/components/dashboard-header/ASDashboardHeader.tsx:26]: change namin in the styles

File: src/components/mastery-Of-The-Day-Card/ASMasteryOftheDayCard.tsx [hunk 39]
```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { Image, Text, View } from 'react-native'
+
+const completedIcon = require('../../assets/icons/complete.png')
```
Comment (pankaj-negi-vs) [inline src/components/mastery-Of-The-Day-Card/ASMasteryOftheDayCard.tsx]: make these constants in the dashboard-constants file

File: src/components/mastery-Of-The-Day-Card/ASMasteryOftheDayCard.tsx [hunk 40]
```diff
@@ -0,0 +1,51 @@
...
+import { styles } from './asMastery-of-the-day-card-style'
+const ChallengeDeatilsCard = ({ masteryCardDetails }: IChallengeDeatilsCardProps) => {
+  const { img, subheading, heading, timmingDetails, iscompleted, isfavourateMarked } =
+    masteryCardDetails
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.imageContainer}>
+        <View style={styles.imageBackgroundContainer}>
+          <Image style={styles.overThinkerGrey} source={img} />
```
Comment (pankaj-negi-vs) [inline src/components/mastery-Of-The-Day-Card/ASMasteryOftheDayCard.tsx]: change the name of the image

File: src/components/mastery-Of-The-Day-Card/ASMasteryOftheDayCard.tsx [hunk 41]
```diff
@@ -0,0 +1,51 @@
...
+        </View>
+      </View>
+      <View style={styles.detailsContainer}>
+        <View style={styles.detailsSubContainer}>
+          <View style={styles.detailsMainContainer}>
```
Comment (pankaj-negi-vs) [inline src/components/mastery-Of-The-Day-Card/ASMasteryOftheDayCard.tsx]: change the name of the container as the main container comes first

File: src/components/mastery-Of-The-Day-Card/ASMasteryOftheDayCard.tsx [hunk 42]
```diff
@@ -0,0 +1,51 @@
...
+
+const completedIcon = require('../../assets/icons/complete.png')
+const markedFavourite = require('../../assets/icons/markedFavourite.png')
+const notmMarkedFavourite = require('../../assets/icons/notMarkedFavourite.png')
+const audio = require('../../assets/icons/audio.png')
+
+import { IChallengeDeatilsCardProps } from '../../constants/dashboard-constants'
+import { styles } from './asMastery-of-the-day-card-style'
+const ChallengeDeatilsCard = ({ masteryCardDetails }: IChallengeDeatilsCardProps) => {
+  const { img, subheading, heading, timmingDetails, iscompleted, isfavourateMarked } =
```
Comment (pankaj-negi-vs) [inline src/components/mastery-Of-The-Day-Card/ASMasteryOftheDayCard.tsx]: correct the spelling of img to image and timming to timing

File: src/components/mastery-Of-The-Day-Card/asMastery-of-the-day-card-style.ts [hunk 43]
```diff
@@ -0,0 +1,108 @@
...
+  completedIcon: {
+    height: Spacing.space_16,
+    width: Spacing.space_16,
+  },
+  listenTitle: {
+    color: COLORS.neutral[500],
+    fontFamily: 'Quicksand-bold',
+    fontSize: Spacing.space_12,
+    height: Spacing.space_18,
+    letterSpacing: 0.02,
```
Comment (pankaj-negi-vs) [inline src/components/mastery-Of-The-Day-Card/asMastery-of-the-day-card-style.ts]: doesn't need it

File: src/components/mastery-Of-The-Day-Card/asMastery-of-the-day-card-style.ts [hunk 44]
```diff
@@ -0,0 +1,108 @@
...
+    height: Spacing.space_16,
+    width: Spacing.space_16,
+  },
+  completedIcon: {
+    height: Spacing.space_16,
+    width: Spacing.space_16,
+  },
+  listenTitle: {
+    color: COLORS.neutral[500],
+    fontFamily: 'Quicksand-bold',
```
Comment (pankaj-negi-vs) [inline src/components/mastery-Of-The-Day-Card/asMastery-of-the-day-card-style.ts]: use typography

File: src/components/top-wreckers/ASTopWrecker.tsx [hunk 45]
```diff
@@ -0,0 +1,35 @@
+import React, { useEffect, useState } from 'react'
+import { ActivityIndicator, Text, View } from 'react-native'
+import WreckerCard from '../wrecker-card/ASWreckerCard'
+
+import { Image } from 'react-native-elements'
+import { getWreckerData } from '../../services/api/get-wrecker-data'
+import { styles } from './asTop-wrecker-style'
+const nextButtonIcon = require('../../assets/icons/next.png')
+const TopWrecker = () => {
+  const [isLoadingWreckerData, setLoadingWreckerData] = useState(true)
```
Comment (pankaj-negi-vs) [inline src/components/top-wreckers/ASTopWrecker.tsx]: define types for <boolean>

File: src/components/top-wreckers/ASTopWrecker.tsx [hunk 46]
```diff
@@ -0,0 +1,35 @@
...
+  const [wreckerData, setWreckerData] = useState([])
```
Comment (pankaj-negi-vs) [inline src/components/top-wreckers/ASTopWrecker.tsx:13]: make interface for this at the top of this file

File: src/components/top-wreckers/ASTopWrecker.tsx [hunk 47]
```diff
@@ -0,0 +1,35 @@
...
+        <Text style={styles.headingText}>Your Top Wreckers</Text>
+      </View>
+      <View style={styles.topWreckers}>
+        {isLoadingWreckerData ? (
+          <ActivityIndicator />
+        ) : (
+          wreckerData.map((data, index) => <WreckerCard key={index} data={data} />)
+        )}
+      </View>
+      <View style={styles.SubdetailsContainer}>
```
Comment (pankaj-negi-vs) [inline src/components/top-wreckers/ASTopWrecker.tsx]: subDetailsContainer

File: src/components/top-wreckers/ASTopWrecker.tsx [hunk 48]
```diff
@@ -0,0 +1,35 @@
...
+        <Text style={styles.SubdetailsText}>View All Wreckers</Text>
```
Comment (pankaj-negi-vs) [inline src/components/top-wreckers/ASTopWrecker.tsx]: subDetailsText

File: src/components/wrecker-card/ASWreckerCard.tsx [hunk 49]
```diff
@@ -0,0 +1,31 @@
...
+import { IWreckerCardProp, imageDetails } from '../../constants/assesment-constants'
+import { styles } from './asWrecker-card-style'
+
+const WreckerCard = (props: IWreckerCardProp) => {
+  const { data } = props
+  const { title, progress, description } = data
+  return (
+    <View style={styles.cardContainer}>
+      <View style={styles.imageContainer}>
+        <Image style={styles.image} source={imageDetails[title]} />
```
Comment (pankaj-negi-vs) [inline src/components/wrecker-card/ASWreckerCard.tsx]: change name to wreckerImage

File: src/constants/dashboard-constants.ts [hunk 50]
```diff
@@ -0,0 +1,68 @@
+import { COLORS } from '../theme/colors'
+
+const overthinkerGrey = require('../assets/icons/overthinkerGrey.png')
```
Comment (pankaj-negi-vs) [inline src/constants/dashboard-constants.ts]: use export to use them in the main files

File: src/constants/dashboard-constants.ts [hunk 51]
```diff
@@ -0,0 +1,68 @@
...
+const dActive = require('../assets/images/dActive.png')
+const reflection = require('../assets/images/reflection.png')
+const reminder = require('../assets/images/reminder.png')
+export interface IChallengeDeatilsCard {
```
Comment (pankaj-negi-vs) [inline src/constants/dashboard-constants.ts]: img and timmingdetails

File: src/constants/dashboard-constants.ts [hunk 52]
```diff
@@ -0,0 +1,68 @@
...
+  img: any
+  subheading: string
+  heading: string
+  timmingDetails: string
+  iscompleted: boolean
+  isfavourateMarked: boolean
+  backgroundColor: string
+}
+
+export interface IChallengeDeatil {
```
Comment (pankaj-negi-vs) [inline src/constants/dashboard-constants.ts]: correct the spelling IChallengeDeatil

File: src/screens/assesment/Assesment.tsx [hunk 53]
```diff
@@ -0,0 +1,66 @@
+import React from 'react'
+import { Text, View } from 'react-native'
+import { Image } from 'react-native-elements'
+import LinearGradient from 'react-native-linear-gradient'
+import { palette } from '../../theme/palette'
+
+import TopWrecker from '../../components/top-wreckers/ASTopWrecker'
+import { styles } from './assesment-style'
+const surviving = require('../../assets/icons/surviving.png')
```
Comment (pankaj-negi-vs) [inline src/screens/assesment/Assesment.tsx]: put these consts in the constant file

File: src/screens/dashboard/Dashboard.tsx [hunk 54]
```diff
@@ -1,97 +1,43 @@
...
     <>
-      <Header/>
-      <View style={styles.MasteryOftheDayCard}>
-
-      <MasteryOftheDayCard masteryCardDetails={masteryCardDetails} />
+      <Header />
+      <View style={styles.masteryOftheDayCard}>
+        <MasteryOftheDayCard masteryCardDetails={masteryCardDetails} />
       </View>
       <View style={styles.flatListContainer}>
```
Comment (pankaj-negi-vs) [inline src/screens/dashboard/Dashboard.tsx:26]: change the name

File: src/utils/common-utils.ts [hunk 55]
```diff
@@ -33,3 +33,27 @@ export const calculateNotificationTime = (timeOfCreation: string): string => {
 
   return notificationTime
 }
+
+export const getTodayDate = (): string => {
```
Comment (pankaj-negi-vs) [inline src/utils/common-utils.ts]: change name for getCurrentDate

**Pull request merged**